src/construction.cpp

```diff
@@ -1130,21 +1134,20 @@ void construct::done_window_curtains( const tripoint & )
         _( "After boarding up the window the curtains and curtain rod are left." ) );
 }
 
-void construct::done_extract_sand( const tripoint &p )
+void construct::done_extract_maybe_revert_to_dirt( const tripoint &p )
 {
-    g->m.spawn_item( g->u.pos(), "material_sand", 1, rng( 300, 600 ) );
     if( one_in( 10 ) ) {
         g->m.ter_set( p, t_dirt );
     }
-    g->u.add_msg_if_player( _( "You gather some sand." ) );
-}
-void construct::done_extract_clay( const tripoint &p )
-{
-    g->m.spawn_item( g->u.pos(), "clay_lump", rng( 6, 12 ) );
-    if( one_in( 10 ) ) {
-        g->m.ter_set( p, t_dirt );
+
+    if( g->m.ter( p ) == t_clay ) {
+        g->u.add_msg_if_player( _( "You gather some clay." ) );
```

Comment:
Calling `add_msg_if_player` on the player object seems kinda pointless. It *is* the player. You might as well call `add_msg` directly.

Reply:
Makes sense. I was just squishing together the existing methods which already called `add_msg_if_player`, but no reason I can't leave it better than I found it if the semantics are the same in this context.